### ISSUE TITLE
Parser error on "combining character" (U+0307)

### ISSUE THREAD

**Issue by skatsman** [google/closure-compiler#3639]
### There seems to be an issue with parsing composite character sequences.

Composite character `İ` consist of the base character and combining character sequence (U+0049)(U+0307)  
Using this character causing the compiler to fail with the parser error.

### Example code

```
var bar = {
İ: "foo"
};
```

### Compiler error

```
input0:2: ERROR - [JSC_PARSE_ERROR] Parse error. Character '̇' (U+0307) is not a valid identifier start char
  2| İ: "foo"
      ^

1 error(s), 0 warning(s)
```

### Demo
[Example](https://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520bar%2520%253D%2520%257B%250AI%25CC%2587%253A%2520%2522foo%2522%250A%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3Dtrue%26ENABLE_ALL_DIAGNOSTIC_GROUPS%3Dtrue%26CHECK_TYPES%3Dtrue%26CHECK_SYMBOLS%3Dtrue%26MISSING_PROPERTIES%3Dtrue%26TRANSPILE%3Dtrue%26CLOSURE_PASS%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue)

**Comment by rrdelaney** [comment]
Is this a blocking error for you?

**Comment by skatsman** [comment]
This prevents compiling one of the dependencies ([lower-case](https://github.com/blakeembrey/change-case/blob/c19866744e0ea07bb167602a81a664162d59618e/packages/lower-case/src/index.ts#L18)) of [graphql-tools](https://github.com/ardatan/graphql-tools) which we use in our project

**Comment by jpduvet** [comment]
Exactly the same issue with (U+3007) preventing compiling the dependency "fast-redact"

**Comment by smogg** [comment]
Same problem with [`@tanstack/react-query-devtools`](https://tanstack.com/query/v4/docs/react/devtools)

**Comment by brancusi** [comment]
Yes, getting this as well when using @tanstack/react-query-devtools

**Comment by SergejGleithmann** [comment]
Hello, any idea, why the stopped working on this? Is there a way to bypass this issue? Im also having this with some leaflet react plugins.

**Comment by jwr** [comment]
Just encountered this, and it is a showstopper here.

Errors encountered while trying to parse file
  [...]/node_modules/lower-case/dist/index.js
  {:message "Character '̇' (U+0307) is not a valid identifier start char", :line 13, :column 13}
